website/docs/reference/resource-properties/deprecation_date.md

```diff
@@ -28,5 +28,46 @@ models:
+
+## Explanation
+
+### Purpose
+
+Declaring a `deprecation_date` for a dbt model provides a mechanism to communicate plans and timelines for long-term support and maintenance and to facilitate change management.
+
+Setting a `deprecation_date` works well in conjunction with other [model governance](/docs/collaborate/govern/about-model-governance) features like [model versions](/docs/collaborate/govern/model-versions), but can also be used independently from them.
+
+### Warning messages
+
+When a project references a model that's slated for deprecation or the deprecation date has passed, a warning is generated. If it's a versioned model, with a newer version available, then the warning says so. This added bit of cross-team communication, from producers to consumers, is an advantage of using dbt's built-in functionality around model versions to facilitate migrations.
+
+Additionally, [`WARN_ERROR_OPTIONS`](/reference/global-configs/warnings) gives a mechanism whereby users can promote these warnings to actual runtime errors:
+- `DeprecatedModel` - warning when parsing a project that defines deprecated model(s)
+- `UpcomingDeprecationReference` - warning when referencing a model with a future deprecation date
+- `DeprecatedReference` - warning when referencing a model with a past deprecation date
+
+### Selection syntax
+
+There is not specific [node selection syntax](/reference/node-selection/syntax) for `deprecation_date`. [Programmatic invocations](/reference/programmatic-invocations) is one way to identify deprecated models (potentially in conjunction with [dbt list](/reference/commands/list)). e.g., `dbt -q ls  --output json --output-keys database schema alias deprecation_date`.
+
+### Deprecation process
+
+Additional steps are necessary to save on build-related compute and storage costs for a deprecated model.
+
+Deprecated models can continue to be built by producers and be selected by consumers until they are [disabled](/reference/resource-configs/enabled) or removed.
+
+Just like it does not automatically [drop relations when models are deleted](/faqs/models/removing-deleted-models), dbt does not drop relations for deprecated models.
+
+Strategies similar to [here](https://discourse.getdbt.com/t/faq-cleaning-up-removed-models-from-your-production-schema/113) or [here](https://discourse.getdbt.com/t/clean-your-warehouse-of-old-and-deprecated-models/1547) can be used to drop relations that have been deprecated and are no longer in use.
+
+### Table expiration on BigQuery
+
+dbt-bigquery can set an [`hours_to_expiration`](/reference/resource-configs/bigquery-configs#controlling-table-expiration) that translates to `expiration_timestamp` within BigQuery.
+
+dbt does not automatically synchronize `deprecation_date` and `hours_to_expiration`, but users may want to coordinate them in some fashion (such as setting a model to expire 48 hours after its `deprecation_date`). Expired tables in BigQuery will be deleted and their storage reclaimed.
```

Comment:
> setting a model to expire 48 hours after its `deprecation_date`

And it's reasonable enough to do this, using the `datetime` module? That's neat!

Reply:
To unblock merging this PR, deferring the actual demonstration of this to a separate issue: #3765